### ISSUE TITLE
Vagrant Help + Broken Vagrant Files

### ISSUE THREAD

**Issue by lloy0076** [hashicorp/vagrant#3180]
TL;DR

If the 'vagrant' command sees 'vagrant help [some_command]' it probably should not fail if a Vagrantfile is malformed for some reason. It doesn't _need_ a Vagrantfile to print out the 'help' documentation.

LONG

I did:

```
Isengaard-2:test lloy0076$ vagrant help
Vagrant failed to initialize at a very early stage:

There was an error loading a Vagrantfile. The file being loaded
and the error message are shown below. This is usually caused by
a syntax error.

Path: .../test/Vagrantfile
Message: uninitialized constant VAGRANTFILE_API_VERSION
Isengaard-2:test lloy0076$ mv Vagrantfile nonVagrantFile
Isengaard-2:test lloy0076$ vagrant help package
Usage: vagrant package [vm-name] [--base name] [--output name.box]
                       [--include one,two,three] [--vagrantfile file]

        --base NAME                  Name of a VM in virtualbox to package as a base box
        --output NAME                Name of the file to output
        --include x,y,z              Additional files to package with the box.
        --vagrantfile file           Vagrantfile to package with the box.
    -h, --help                       Print this help
```

Maybe Vagrant should complain about the misconfigured Vagrantfile -- that error is legitimate -- but this just bamboozles me.

A few possible suggestions:
1. If _vagrant help [something]_ is the command, print the help and then any other possible errors because of the Vagrantfile in same directory.
2. More generally, maybe the error should also print out what it appears the user was trying to do

Let me explain.

Let's say that my wondrous building script automates Vagrant but I muddle a Vagrantfile up. Let's just say that I also -- for reasons of incompetence -- do not log what Vagrant commands I executed.

And then Vagrant falls over with:

```
Vagrant failed to initialize at a very early stage:

There was an error loading a Vagrantfile. The file being loaded
and the error message are shown below. This is usually caused by
a syntax error.

Path: .../test/Vagrantfile
Message: uninitialized constant VAGRANTFILE_API_VERSION
```

I know which Vagrantfile it is but not what command; it would be somewhat nice if Vagrant said:

```
Vagrant failed to initialize at a very early stage:
You tried to run: vagrant up

There was an error loading a Vagrantfile. The file being loaded
and the error message are shown below. This is usually caused by
a syntax error.

Path: .../test/Vagrantfile
Message: uninitialized constant VAGRANTFILE_API_VERSION
```
